### DISCUSSION TITLE
BasicVSR++: reproduce ntire decompression results.

### DISCUSSION THREAD

**Discussion by kai422** [open-mmlab/mmediting#1272] [Models] [answered]
Thanks for your great work!
I try to reproduce results of ntire track-3 with provided pretrained weight. 
I filled **lq_folder** with  _NTIRE21_validation_fixed-rate_ and  **gt_folder** with _NTIRE21_validation__raw_,
Use _basicvsr_plusplus_c128n25_ntire_decompress_track3_20210304-6daf4a40.pth_ , the final testing PSNR for the validation set is 28.67.
The testing script I used is

> python tools/test.py configs/restorers/basicvsr_plusplus/basicvsr_plusplus_c128n25_600k_ntire_decompress_track3.py weights/basicvsr_plusplus_c128n25_ntire_decompress_track3_20210304-6daf4a40.pth

I wonder :
1. if it's possible for you to provide the results for ntire track3 validation set?
2. How could I achieve PSNR result to the same level with the result 30.37 of the test set which is reported in the ntire workshop paper?

Are there any other steps I need to follow?
Thanks!

**Comment by ckkelvinchan** [comment]
30.37 is the PSNR on the test set, which is not the same as the validation set. Only the organizer has the GT of the test set.

This model is the model we used for the competition.

**Comment by kai422** [comment]
May I check for the test set, do I need to specify additional parameters in the testing pipeline for ensemble (×8) testing? (including Flip and Rotation). Thanks!

**Comment by ckkelvinchan** [comment]
> May I check for the test set, do I need to specify additional parameters in the testing pipeline for ensemble (×8) testing? (including Flip and Rotation). Thanks!

So far we did not implement it. You can modify the [basicvsr.py](https://github.com/open-mmlab/mmediting/blob/master/mmedit/models/restorers/basicvsr.py) for the ensemble. I will add support to it later. Thank you very much for letting us know your need :)

**Comment by kai422** [comment]
Thank you!

**Comment by kai422** [comment]
For the validation set, the PSNR for unprocessed video is 27.09, the 28.67(+1.58) from basicvsr++ is of the same improvement level as the test set. (no ensemble testing)

In the nitire report,  for the test set, the unprocessed video PSNR is 28.34 compared to 30.37(+2.03)  of basicvsr++. (8x ensemble testing).

Thanks for your help!

**Comment by Kadakol** [comment]
Hi @kai422 and @ckkelvinchan ,

Can I ask how to get access to the validation raw data? The [NTIRE challenge](https://competitions.codalab.org/competitions/28035#participate) only seems to be providing the compressed data for validation. Am I missing something here? Please suggest.

![image](https://user-images.githubusercontent.com/6085078/133572709-d9cadb62-37da-4efb-b89d-32835e8a65ca.png)

**Comment by kai422** [comment, marked as answer]
> Hi @kai422 and @ckkelvinchan ,
> 
> Can I ask how to get access to the validation raw data? The [NTIRE challenge](https://competitions.codalab.org/competitions/28035#participate) only seems to be providing the compressed data for validation. Am I missing something here? Please suggest.
> 
> ![image](https://user-images.githubusercontent.com/6085078/133572709-d9cadb62-37da-4efb-b89d-32835e8a65ca.png)

Please check this: https://github.com/RenYang-home/NTIRE21_VEnh

**Comment by Kadakol** [comment]
Thank you so much @kai422 !
